File: demo/CirclePicker/ViewController.m
//
//  ViewController.m
//  CirclePicker
//
//  Created by vladyslav.anokhin on 9/22/15.
//  Copyright (c) 2015 Appus. All rights reserved.
//

#import "ViewController.h"
#import "CircularPickerView.h"

@interface ViewController ()<UITextFieldDelegate>
@property (weak, nonatomic) IBOutlet UILabel *valueLabel;
@property (weak, nonatomic) IBOutlet UILabel *lineWidthLabel;


@property (weak, nonatomic) IBOutlet CircularPickerView *picker;
@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    self.lineWidthLabel.text = @(self.picker.lineWidth).stringValue;
    self.valueLabel.text     = @(self.picker.defaultValue).stringValue;
}

#pragma mark - Actions

- (IBAction)switchChanged:(UISwitch *)sender {
    self.picker.autocomplete = sender.isOn;
}


- (IBAction)vladuChanged:(CircularPickerView *)sender {
    self.valueLabel.text = [NSString stringWithFormat:@"%.2f",sender.currentValue];
}


- (IBAction)lineWidthChanged:(UIStepper *)sender {
     self.lineWidthLabel.text = @(sender.value).stringValue;
     self.picker.lineWidth = sender.value;
}


@end

File: demo/CircularPickerViewSwift/ViewController.swift
//
//  ViewController.swift
//  CircularPickerViewSwift
//
//  Created by Sergey Sokoltsov on 12/14/16.
//  Copyright © 2016 Sergey Sokoltsov. All rights reserved.
//

import UIKit
import Foundation
import AppusCircularPickerView

class ViewController: UIViewController {
    @IBOutlet weak var circularPicker : AppusCircularPickerView?
    @IBOutlet weak var valueLabel : UILabel?
    @IBOutlet weak var lineWidthStepper : UIStepper?
    @IBOutlet weak var lineWidthLabel : UILabel?

    override func viewDidLoad() {
        super.viewDidLoad()

        valueLabel?.text = String(format:"%.2f", circularPicker!.currentValue)
        lineWidthLabel?.text = String(format:"%i", NSInteger(circularPicker!.lineWidth))
        lineWidthStepper?.value = Double(circularPicker!.lineWidth)
    }

    @IBAction func switchChanged(sender: UISwitch) {
        circularPicker?.autocomplete = sender.isOn
    }

    @IBAction func circularPickerValueChanged(_ sender: AppusCircularPickerView) {
        valueLabel?.text = String(format:"%.2f", sender.currentValue)
    }

    @IBAction func lineWidthChanged(sender: UIStepper) {
        lineWidthLabel?.text = String(format:"%i", NSInteger(sender.value))
        circularPicker?.lineWidth = Float(sender.value)
    }
}

